docs/GettingStarted.md

```diff
@@ -132,11 +98,10 @@ firstName
 [...]
 ```
 
-The generator supports three different types of constraint. These are:
+The generator supports two different types of constraint:
 
--   **Predicates** - boolean-valued functions that define whether a given value is valid or invalid. In the above profile `ofType` and `matchingRegex` are examples of predicates.
--   **Grammatical** - combine or modify other constraints. They are fully recursive; any grammatical constraint is a valid input to any other grammatical constraint. In the above profile `not` is an example of a grammatical constraint.
--   **Presentational** - provide additional formatting information, we'll cover these later.
+-   [**Predicates**](GettingStarted.md#Data-Types) - boolean-valued functions that define whether a given value is valid or invalid. In the above profile `ofType` and `matchingRegex` are examples of predicates.
+-   [**Grammatical**](GettingStarted.md#Data-Types) - combine or modify other constraints including other grammatical constraints. In the above profile `not` is an example of a grammatical constraint.
 
 The current profile outputs random text strings for the `firstName` field. Depending on what you are intending to use this data for this may or may not be appropriate. For testing purposes, you are likely to want output data that has a lot of variability, however, if you are using the generator to create simulation data, you might want to use a regex that creates more realistic data, for example the regex `(Joh?n|Mar[yk])`.
```

Comment:
This guide was written before the `firstname` type was added. Can you please elaborate the example, perhaps (1) generate using the `string` type, (2) demonstrate a regex constraint, (3) introduce the `firstname` type, (4) perhaps show an interesting constraint such as `longerThan` - here's [an example](https://colineberhardt.github.io/datahelix/playground/#ewogICJzY2hlbWFWZXJzaW9uIjogIjAuMTgiLAogICJmaWVsZHMiOiBbCiAgICB7ICJuYW1lIjogIm5hbWUiLCAidHlwZSI6ICJmaXJzdG5hbWUiIH0sCiAgICB7ICJuYW1lIjogImFnZSIsICJ0eXBlIjogImludGVnZXIiIH0KICBdLCAKICAiY29uc3RyYWludHMiOiBbCiAgICB7CiAgICAgICJmaWVsZCI6ICJuYW1lIiwKICAgICAgImxvbmdlclRoYW4iOiAxMAogICAgfSwKICAgIHsKICAgICAgImZpZWxkIjogImFnZSIsCiAgICAgICJncmVhdGVyVGhhbk9yRXF1YWxUbyI6IDAKICAgIH0sCiAgICB7CiAgICAgICJmaWVsZCI6ICJhZ2UiLAogICAgICAibGVzc1RoYW4iOiAxMDAKICAgIH0KICBdCn0KICAgICAgICA%3D)

Reply:
I've slightly changed the example we go through in the docs to be a little more straight forward. I want to add in a small section on faker when that goes in but let me know if you think I need to add any more constraints.

Reply:
Some key things that are missing are inter field constrants, inMap constraints and grammatical constraints